scripts/create_gitlab_snapshot.py

```diff
@@ -0,0 +1,60 @@
+import datetime
+import logging
+from more_itertools import (
+    one,
+)
+
+from azul import (
+    config,
+    require,
+)
+from azul.deployment import (
+    aws,
+)
+from azul.logging import (
+    configure_script_logging,
+)
+
+from azul.terraform import (
+    _adjust_name_tag,
+    _tags,
+)
+
+logger = logging.getLogger(__name__)
+
+
+def create_snapshot():
+    require(config.terraform_component == 'gitlab',
+            'Please _select the GitLab environment for the targeted deployment')
+    filters_ = [
+        {
+            'Name': 'key-name',
+            'Values': ['azul-gitlab']
+        }
+    ]
+    instance = one([
+        one(reservation['Instances'])
+        for reservation in aws.ec2.describe_instances(Filters=filters_)['Reservations']
+    ])
+    assert instance['State']['Name'] == 'stopped', instance['State']
+    resource_tags = [
+        {'Key': k, 'Value': v}
+        for k, v in _adjust_name_tag('aws_gitlab',
+                                     _tags('gitlab', name='azul_gitlab')).items()
+    ]
+    operator_name = aws.sts.get_caller_identity()['UserId'].split(':')[1]
```

Comment:
This seems brittle. How do you know for sure what's in the second ARN component? We should have more robust code for this in the `azul.deployment` module.

Reply:
https://docs.aws.amazon.com/IAM/latest/UserGuide/reference_policies_variables.html#principaltable

Rather than splitting down to the second component of the UserID, I'll just include the whole thing. It will be in form of `role-id:caller-specified-role-name`.